openapi_python_client/templates/property_templates/union_property.pyi

```diff
@@ -19,17 +30,12 @@ def _parse_{{ property.python_name }}(data: {{ property.get_type_string(json=Tru
         return {{ property.python_name }}
     except: # noqa: E722
         pass
-    {% else  %}{# Don't do try/except for the last one #}
-    {% from "property_templates/" + inner_property.template import construct, check_type_for_construct %}
-    if not {{ check_type_for_construct("data") }}:
-        raise TypeError()
-    {{ construct(inner_property, "data", initial_value="UNSET") | indent(4) }}
-    return {{ property.python_name }}
-    {% endif %}
     {% endfor %}
     {% if property.has_properties_without_templates %}
     {# Doesn't really matter what we cast it to as this type will be erased, so cast to one of the options #}
     return cast({{ property.get_type_string() }}, data)
+    {% else %}
```

Comment:
Do one more check to see if it's a Dict and then return UnknownType in that case, fall back to raw otherwise